### DISCUSSION TITLE
Get DAG_ID as task parameter

### DISCUSSION THREAD

**Discussion by romanzdk** [apache/airflow#34479] [Q&A] [answered]
Hi, is it possible to somehow get dag_id into some Operator e.g. like
```python
def CustomOperator(**kwargs):
	kwargs['labels'] = {
		'runner': 'airflow',
		'dag_id': ??,
	}
	return KubernetesPodOperator(**kwargs)

with DAG(
	dag_id = 'some-dag-id',
	start_date = datetime.datetime(2023, 9, 16),
	default_args = default_args,
):
	some_pod = CustomOperator(task_id = 'some_name')
```

**Comment by Taragolis** [comment, marked as answer]
If you do not specify it in task explicitly you can't get it because in this case `dag_id` will assign in the end of DAG contexts manager.

Also post here suggestion from the slack thread:

[Jinja Templating](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/operators.html#concepts-jinja-templating)? labels attribute is templated in KubernetesPodOperator so you could use context variable from [Templates reference](https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html)

```python
with DAG(
    dag_id = 'some-dag-id',
    start_date = datetime.datetime(2023, 9, 16),
    default_args = default_args,
):
    some_pod = KubernetesPodOperator(
        task_id = 'some_name'
        labels = {
            'runner': "airflow",
            'dag_id': "{{ dag.dag_id }}",
        }
        ...
)
```
